server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java

```diff
@@ -156,21 +147,14 @@ private AsyncSingleAction(Request request, ActionListener<Response> listener) {
                 throw blockException;
             }
 
-            String concreteSingleIndex;
-            if (resolveIndex(request)) {
-                concreteSingleIndex = IndexNameExpressionResolver.concreteSingleIndex(clusterState, request).getName();
```

Comment:
Has the index name in the request already been resolved earlier?

Reply:
Apparently it's always called with the right index names - not aliases.

This was effectively dead code. All the places where `resolveIndex` was true never accessed the `concreteIndex` property of the `InternalRequest`. There was one place where `concreteIndex()` was used, but that had `resolveIndex()` overriden to return false.